keyboards/dmqdesign/spin/keymaps/encoderLayers/keymap.c

```diff
@@ -0,0 +1,151 @@
+                    midi_send_cc(&midi_device, 0, 0x14, 1);
+                } else {
+                    midi_send_cc(&midi_device, 0, 0x15, 1);
+                }
+                break;
+            case _TL:
+                if (clockwise) {
+                    midi_send_cc(&midi_device, 0, 0x1A, 1);
+                } else {
+                    midi_send_cc(&midi_device, 0, 0x1B, 1);
+                }
+                break;             
+        }
+    } else if (index == 1) { /* Second encoder */
+        switch (currentLayer) {
+            case _BL:
+                if (clockwise) {
+                    tap_code(KC_VOLU);
+                } else {
+                    tap_code(KC_VOLD);
+                }
+                break;
+            case _FL:
+                if (clockwise) {
+                    midi_send_cc(&midi_device, 0, 0x16, 1);
+                } else {
+                    midi_send_cc(&midi_device, 0, 0x17, 1);
+                }
+                break;
+            case _TL:
+                if (clockwise) {
+                    midi_send_cc(&midi_device, 0, 0x1C, 1);
+                } else {
+                    midi_send_cc(&midi_device, 0, 0x1D, 1);
+                }
+                break;
+        }
+    } else if (index == 2) { /* Third encoder */
+        switch (currentLayer) {
+            case _BL:
+                if (clockwise) {
+                    rgblight_increase_val();
+                } else {
+                    rgblight_decrease_val();
+                }
+                break;
+            case _FL:
+                if (clockwise) {
+                    midi_send_cc(&midi_device, 0, 0x18, 1);
+                } else {
+                    midi_send_cc(&midi_device, 0, 0x19, 1);
+                }
+                break;
+            case _TL:
+                if (clockwise) {
+                    midi_send_cc(&midi_device, 0, 0x1E, 1);
+                } else {
+                    midi_send_cc(&midi_device, 0, 0x1F, 1);
+                }
+                break;
+        }
+    }
+}
+
+layer_state_t layer_state_set_user(layer_state_t state) { //This will run every time the layer is updated
+    currentLayer = get_highest_layer(state);
+
+    switch (currentLayer) {
+        case _BL:
+            setrgb(RGB_WHITE, (LED_TYPE *)&led[0]); //Set the top LED to white for the bottom layer
+            setrgb(0, 0, 0, (LED_TYPE *)&led[1]);
+            setrgb(0, 0, 0, (LED_TYPE *)&led[2]);
+            break;
+        case _FL:
+            setrgb(0, 0, 0, (LED_TYPE *)&led[0]); //Set the middle LED to white for the middle layer
+            setrgb(RGB_WHITE, (LED_TYPE *)&led[1]);
+            setrgb(0, 0, 0, (LED_TYPE *)&led[2]);
+            break;
+        case _TL:
+            setrgb(0, 0, 0, (LED_TYPE *)&led[0]);
+            setrgb(0, 0, 0, (LED_TYPE *)&led[1]);
+            setrgb(RGB_WHITE, (LED_TYPE *)&led[2]); //Set the bottom LED to white for the top layer
```

Comment:
You shouldn't need the casts here; `led` is already of `LED_TYPE`.

Reply:
removed the unnecessary typecasts.